rpc/handlers.go

```diff
@@ -1229,10 +1229,10 @@ func (h *Handler) SimulateTransactions(id BlockID, transactions []BroadcastedTra
 	return result, nil
 }
 
-func (h *Handler) TraceBlockTransactions(blockHash felt.Felt) ([]TracedBlockTransaction, *jsonrpc.Error) {
-	block, err := h.bcReader.BlockByHash(&blockHash)
-	if err != nil {
-		return nil, ErrInvalidBlockHash
+func (h *Handler) TraceBlockTransactions(id BlockID) ([]TracedBlockTransaction, *jsonrpc.Error) {
+	block, err := h.blockByID(&id)
+	if block == nil || err != nil {
+		return nil, ErrBlockNotFound
```

Comment:
Even as the spec still says there should be "invalid block hash" error I think it doesn't make sense anymore. Will propose this change to the spec.